bugbug/similarity.py

```diff
@@ -175,3 +180,111 @@ def get_similar_bugs(self, query):
         return [
             self.bug_ids[ind] for ind in indices[0] if self.bug_ids[ind] != query["id"]
         ]
+
+
+class Word2VecWmdSimilarity(BaseSimilarity):
+    def __init__(self):
+        self.corpus = []
+        self.bug_ids = []
+        for bug in bugzilla.get_bugs():
+            self.corpus.append(text_preprocess(get_text(bug)))
+            self.bug_ids.append(bug["id"])
+        self.w2vmodel = Word2Vec(self.corpus, size=100, min_count=5)
+        self.w2vmodel.init_sims(replace=True)
+
+    def wmdistance(self, document1, document2, all_distances, distance_metric="cosine"):
```

Comment:
Can you check the differences between this function and the equivalent present in gensim? I don't remember exactly why I rewrote it.

Reply:
There is a subtle difference. Here we've computed the cosine distances before calling wmdistance and stored them in all_distances. (and then we use this inside the wmdistance method.). On the other hand the word2vec [wmdistance](https://tedboy.github.io/nlps/_modules/gensim/models/word2vec.html#Word2Vec.wmdistance) calculates them inside the method. 

The other major difference is that `word2vec.wmdistance` calculates only the euclidean distance. (There isn't an option to calculate cosine distances.), so we need to implement it separately over here.

Reply:
Can you add a comment explaining that the function partially comes from gensim and explaining the differences?